collector/lib/ProcfsScraper.cpp

```diff
@@ -349,9 +349,35 @@ bool ReadConnectionsFromFile(Address::Family family, L4Proto l4proto, std::FILE*
   return true;
 }
 
-// GetConnections reads all active connections (inode -> connection info mapping) for a given network NS, addressed by
+// ReadUdpEndpointsFromFile reads all udp endpoints from a `net/udp[6]` file and stores them by inode in the given map.
+bool ReadUdpEndpointsFromFile(Address::Family family, std::FILE* f,
```

Comment:
Perhaps I should not have separate functions for reading UDP and TCP endpoints. For now this seemed like the safest solution.

Reply:
Good idea. I took this even a step further and I put that for loop you proposed in its own function. That way it is easier to write unit tests for it.